src/driver-sinowealth.c

```diff
@@ -1796,6 +1796,16 @@ sinowealth_update_config_from_profile(struct ratbag_profile *profile)
 	ratbag_profile_for_each_resolution(profile, resolution) {
 		if (!resolution->dpi_x || !resolution->dpi_y)
 			continue;
+
+		/* Limit the resolution if it somehow got higher than allowed. */
+		{
```

Comment:
don't think we need the extra block here, we support C99 anyway and we can probably start using late declarations

Reply:
I've actually put it in a block not to pollute the namespace in this scope (personal preference). If you don't think it makes sense I will remove it. :)